src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -614,86 +753,29 @@ void LFUDAPolicy::cleaning(const DoutPrefixProvider* dpp)
       block.size = 0;
       block.blockID = 0;
       if (c_obj->have_instance()) {
-        blockDir->get(dpp, &block, null_yield);
-        if (block.version == c_obj->get_instance()) { //versioned case - update head block entry that has latest version
-          op_ret = blockDir->update_field(dpp, &block, "dirty", "false", null_yield);
-          if (op_ret < 0) {
-              ldpp_dout(dpp, 20) << __func__ << "updating dirty flag in block directory for head failed!" << dendl;
-          }
-        }
+	blockDir->get(dpp, &block, null_yield);
+	if (block.version == c_obj->get_instance()) { //versioned case - update head block entry that has latest version
+	  op_ret = blockDir->update_field(dpp, &block, "dirty", "false", null_yield);
+	  if (op_ret < 0) {
+	      ldpp_dout(dpp, 20) << __func__ << "updating dirty flag in block directory for head failed!" << dendl;
+	  }
+	}
       } else { //non-versioned case
-        op_ret = blockDir->update_field(dpp, &block, "dirty", "false", null_yield);
-        if (op_ret < 0) {
-            ldpp_dout(dpp, 20) << __func__ << "updating dirty flag in block directory for head failed!" << dendl;
-        }
+	op_ret = blockDir->update_field(dpp, &block, "dirty", "false", null_yield);
+	if (op_ret < 0) {
+	    ldpp_dout(dpp, 20) << __func__ << "updating dirty flag in block directory for head failed!" << dendl;
+	}
       }
       if (c_obj->have_instance()) {
-        rgw::d4n::CacheBlock instance_block;
```

Comment:
Your rebase here has again deleted all the code that I added to handle versioned objects, please restore the code correctly.